### PR TITLE
[SW-1644] PySparkling fails to parse spark version ending with +hadoop2.6

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: py/src/ai/h2o/sparkling/VersionComponents.py [hunk 1]
```diff
@@ -33,12 +33,12 @@ def parseFromSparklingWaterVersion(version):
 
     @staticmethod
     def parseFromPySparkVersion(version):
-        match = re.search(r"^((\d+)\.(\d+))\.(\d+)(\.([0-9A-Za-z]+))?$", version)
+        match = re.search(r"^((\d+)\.(\d+))\.(\d+)([\\.\\+]+([0-9A-Za-z\\+\\.]+))?$", version)
```
Comment (mn-mikke) [inline py/src/ai/h2o/sparkling/VersionComponents.py]: WDYT about simplifying the part `([\\.\\+]+([0-9A-Za-z\\+\\.]+))?` of the regular expression to `(.+)?`? Is the suffix used somewhere in the code? Can the suffix start with delimiter `.`,`+`?

I'm just wondering how to generalize the solution for all possible suffixes of spark custom builds ( '-SNAPSHOT', '.cloudera3', etc.)

Comment (jakubhava) [reply py/src/ai/h2o/sparkling/VersionComponents.py]: That's good point, we don't use the suffix at all at this moment

Comment (jakubhava) [reply py/src/ai/h2o/sparkling/VersionComponents.py]: Maybe we can just collect anything after the patch version as suffix, what do you think?

Comment (mn-mikke) [reply py/src/ai/h2o/sparkling/VersionComponents.py]: Sounds good to me!

**Comment by jakubhava** [comment]
@mn-mikke please have a look, the regex should be generalized

**Pull request merged**